src/Moq/Protected/ProtectedMock.cs

```diff
@@ -510,16 +499,62 @@ private static Type[] ToArgTypes(object[] args)
 			return types;
 		}
 
-		private static Expression ToExpressionArg(Type type, object arg)
+		private static bool IsItRefAny(Expression expression)
 		{
-			if (arg is LambdaExpression lambda)
+			return ItRefAnyField(expression) != null;
+		}
+
+		private static FieldInfo ItRefAnyField(Expression expr)
+		{
+			FieldInfo itRefAnyField = null;
+
+			if (expr.NodeType == ExpressionType.MemberAccess)
 			{
-				return lambda.Body;
+				var member = (MemberExpression)expr;
+				if (member.Member is FieldInfo field)
+				{
+					if (field.Name == nameof(It.Ref<object>.IsAny))
+					{
+						var fieldDeclaringType = field.DeclaringType;
+						if (fieldDeclaringType.IsGenericType)
+						{
+							var fieldDeclaringTypeDefinition = fieldDeclaringType.GetGenericTypeDefinition();
+							if (fieldDeclaringTypeDefinition == typeof(It.Ref<>))
+							{
+								itRefAnyField = field;
+							}
+						}
+					}
+				}
 			}
 
+			return itRefAnyField;
+		}
+
+		private static Expression ToExpressionArg(Type type, object arg)
+		{
 			if (arg is Expression expression)
 			{
-				return expression;
+				if (!type.IsAssignableFrom(expression.GetType()))
+				{
+					if(arg is LambdaExpression lambda)
+					{
+						expression = lambda.Body;
+					}
+					return expression;
+				}
+
+				var requiresExpressionMatch = type.IsAssignableFrom(typeof(MethodCallExpression));
+				if (requiresExpressionMatch && expression.IsMatch(out _))
+				{
```

Comment:
Not sure if this is correct. IIRC matchers aren't restricted to `MethodCallExpression` (though that is the most common case). I think we should drop the `requiresExpressionMatch` check.

(Detail: It would be preferable to do the `IsMatch` check as late as possible, since it's the slowest test due to expression compilation.)